server/neptune/gateway/event/source/tmp_working_dir.go

```diff
@@ -0,0 +1,106 @@
+package source
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const workingDirPrefix = "repos"
+
+// TmpWorkingDir handles a tmp workspace on disk for running commands.
+type TmpWorkingDir interface {
```

Comment:
so this is being imported as `source.TmpWorkingDir`? that doesn't really make sense intuitively to me. Also what is source?  And does the caller need to know this is temporary? What the caller really needs to know is that this directory represents a Local Repo right? These are the things to think about when naming imo.

also why this interface? Can we stop exporting interfaces at the sender unless we absolutely need to?